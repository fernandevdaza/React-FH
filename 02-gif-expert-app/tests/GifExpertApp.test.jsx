const { render, screen } = require('@testing-library/react');
const { GifExpertApp } = require('../src/GifExpertApp');

describe('Tests on <GifExpertApp />', () => {
    test('should match with snapshot', () => {
        const { container } = render(<GifExpertApp />);
        expect(container).toMatchSnapshot();
    });
    test('should render h1 with text GifExpertApp', () => {
        render(<GifExpertApp />);
        expect(screen.getByRole('heading', { level: 1 })).toBeTruthy();
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain('GifExpertApp');
    });
});
