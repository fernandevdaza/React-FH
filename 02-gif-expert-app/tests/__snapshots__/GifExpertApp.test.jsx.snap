// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifExpertApp /> should match with snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar Gifs"
      type="text"
      value=""
    />
  </form>
  <h3>
    Nirvana
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
